pkg/controller/events.go

```diff
@@ -133,9 +134,11 @@ func (c *Controller) alert(canary *flaggerv1.Canary, message string, metadata bo
 			}
 			if address, ok := secret.Data["address"]; ok {
 				url = string(address)
+			} else if token, ok := secret.Data["apiToken"]; ok {
```

Comment:
Remove the else from here, people can put the address in the secret or in-line like before, we only need to read the apiToken.

Reply:
@stefanprodan You were right about the previous state of validation, it was shortsighted to break existing configurations. Now I changed it to utilize `anyOf`. Which should be both _not breaking_ for existing clients and allow setting the `address` property in plain text for `slack-api` so secret can be used for token only, therefore we could keep this else block as it is. I believe it is better to keep non sensitive information like API endpoint outside of the secret, but if you insist otherwise I could go back to what you requested earlier.